### DISCUSSION TITLE
ofxRobotArm for UR5e

### DISCUSSION THREAD

**Discussion by jaimie-an** [CreativeInquiry/ofxRobotArm#34] [General]
Hello @madelinegannon, @danzeeeman,

I came across this while researching for better way to work with a UR5e we recently got at my school (I am an MFA student with some coding experience). I thought this would be a great app to include in my project but I am having some trouble.

It seems the beta version doesn't build at all for me. So I tried downloading the master(which seems to have been tested with UR robot rather than ABB anyways). I was able to at least get it up and running, and even connect it to my robot and read the joint positions realtime. But the virtual robots do not update at all, nor does even the preview arm follow the gizmo.

Any help/advice would be greatly appreciated.

Thank you,
Jaimie An

**Comment by danzeeeman** [comment]
project is no longer actively being developed.  I also don't have access to
an e type to try and debug it.

On Fri, Nov 24, 2023 at 3:14 PM jaimie-an ***@***.***> wrote:

> Hello @madelinegannon <https://github.com/madelinegannon>, @danzeeeman
> <https://github.com/danzeeeman>,
>
> I came across this while researching for better way to work with a UR5e we
> recently got at my school (I am an MFA student with some coding
> experience). I thought this would be a great app to include in my project
> but I am having some trouble.
>
> It seems the beta version doesn't build at all for me. So I tried
> downloading the master(which seems to have been tested with UR robot rather
> than ABB anyways). I was able to at least get it up and running, and even
> connect it to my robot and read the joint positions realtime. But the
> virtual robots do not update at all, nor does even the preview arm follow
> the gizmo.
>
> Any help/advice would be greatly appreciated.
>
> Thank you,
> Jaimie An
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CreativeInquiry/ofxRobotArm/discussions/34>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFPVTBUM4JQHPCWCBFY7DTYGD52FAVCNFSM6AAAAAA7ZQ74HWVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZVHA4DOOBTGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Dan Moore
makeitdoathing.com
@theDANtheMAN
***@***.***

**Reply by jaimie-an** [reply]
Thank you for your response. 

I understand that UR5 and UR5e are slightly different. I guess I'm trying to get the baseline code working before adapting it for UR5e.

Initially, even before connecting to the actual robot, the preview arm wasn't even moving to follow the gizmo. I was able to trace the issue to SafetyCheck function in UR5Controller, where '1/60' was being passed in for aDeltaTimeF, which the system as reading as '0' and needed to be updated to '1.0/60.0f'. 

I am now at a point where it can read and position the model to match the real robot, and the preview arm will follow the gizmo. However, now I'm at a point the program as calculating different joint angles given the same TCP point it is reading from the robot. 

I understand that this project is no longer active and you may not be able to help me, but thank you responding anyways!!

Best,
Jaimie

**Comment by danzeeeman** [comment]
Can you push your updates and I’ll try and take a look. Intel Mac? Or what?

**Reply by jaimie-an** [reply]
I have a M1 Mac pro.

I just pushed the updates. However, I should note that, the kinematic model (drawn from ofxURDriver) do not render in correct orientations without some simple updates. For some reason meshs[1] is empty, and the index has to increase by 1 except for [0]. And I'm not sure why it was originally rotating the meshs so much.
In UR5KinematicModel.cpp::draw():
.
.
<img width="599" alt="image" src="https://github.com/CreativeInquiry/ofxRobotArm/assets/143537977/d24188dc-86e0-4d55-a247-518ca91d960a">

**Comment by danzeeeman** [comment]
I need to look at recompiling boost and the rust IK lib dependency now that
OF has stripped some of the conflicts out in 0.12
